File: competitive_programming/leetcode/11_golang/9_palindrome_number.go
package main

import "fmt"

func palindromeNumber(x int) bool {
	if x < 0 {
		return false
	}
	digits := fmt.Sprintf("%d", x)
	for i, j := 0, len(digits)-1; i < j; i, j = i+1, j-1 {
		if digits[i] != digits[j] {
			return false
		}
	}
	return true
}

func main() {
	if !palindromeNumber(121) || palindromeNumber(-121) || palindromeNumber(10) {
		panic("palindrome number failed")
	}
	fmt.Println("9 palindrome number ok")
}

File: competitive_programming/leetcode/09_rust/9_palindrome_number.rs
fn palindrome_number(x: i32) -> bool {
    if x < 0 { return false; }
    let digits: Vec<char> = x.to_string().chars().collect();
    digits.iter().eq(digits.iter().rev())
}

fn main() {
    assert!(palindrome_number(121));
    assert!(!palindrome_number(-121));
    assert!(!palindrome_number(10));
    println!("9 palindrome number ok");
}
